src/main/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImpl.java

```diff
@@ -33,4 +44,42 @@ public CourtHearing prisonToCourt(final Long bookingId, final Long courtCaseId,
     public CourtHearings getCourtHearings(final Long bookingId, final LocalDate fromDate, final LocalDate toDate) {
         return courtHearingsService.getCourtHearingsFor(bookingId, fromDate, toDate);
     }
+
+    @Override
+    public OffenderBooking moveToCell(final Long bookingId, final Long livingUnitId, final String reasonCode, final LocalDateTime dateTime) {
+        log.debug("Received moveToCell request for booking id {}, cell location {}, reasonCode {}, date/time {}",
+                bookingId,
+                livingUnitId,
+                reasonCode,
+                dateTime != null ? dateTime.format(ISO_DATE_TIME) : "null");
+
+        validateMoveToCellRequest(reasonCode);
+        final var movementDateTime = dateTime != null ? dateTime : LocalDateTime.now();
+
+        // TODO DT-235 Just done enough here to write tests for the API - remove once MovementUpdateService has been implemented
+
+        if (bookingId == 123L) {
+            throw new EntityNotFoundException("Simulating a not found for bookingId 123");
+        }
+
+        if (bookingId == 456L) {
+            throw new RuntimeException("Simulating a server error");
+        }
+
+        if (livingUnitId == 123L) {
+            throw new EntityNotFoundException("Simulating a not found for livingUnitId 123");
+        }
+
+        if (reasonCode.equals("123")) {
+            throw new EntityNotFoundException("Simulating a not found for reasonCode '123'");
+        }
+
+        return movementUpdateService.moveToCell(bookingId, livingUnitId, reasonCode, movementDateTime);
+    }
+
+    private void validateMoveToCellRequest(final String reasonCode) {
+        if (StringUtils.isNullOrEmpty(reasonCode)) {
```

Comment:
Not sure it is standard (is anything) but Guava Preconditions are useful for runtime exceptions.  Just a slightly lighter touch.